input/elasticapm/internal/modeldecoder/modeldecoderutil/slice.go

```diff
@@ -17,28 +17,27 @@
 
 package modeldecoderutil
 
-// Reslice increases the slice's capacity, if necessary, to match n.
-// If specified, the newFn function is used to create the elements
-// to populate the additional space appended to the slice.
-func Reslice[Slice ~[]model, model any](slice Slice, n int, newFn func() model) Slice {
-	if diff := n - cap(slice); diff > 0 {
-		// start of the extra space
-		idx := cap(slice)
+import (
+	"slices"
+)
 
-		// Grow the slice
-		// Note: append gives no guarantee on the capacity of the resulting slice
-		// and might overallocate as long as there's enough space for n elements.
-		slice = append([]model(slice)[:cap(slice)], make([]model, diff)...)
-		if newFn != nil {
-			// extend the slice to its capacity
-			slice = slice[:cap(slice)]
+// Reslice returns a slice with length n. It will reallocate the
+// slice if the capacity is not enough.
+func Reslice[Slice ~[]model, model any](slice Slice, n int) Slice {
```

Comment:
[**For reviewers**] I am not a fan of having such util methods but I didn't want to be too disruptive here. Let me know if anyone has a better idea.

Reply:
question: can we maintain the original signature of the Reslice function to reduce the changes required? Apparently we never use Reslice and PopulateNil separately, keeping it down to 1 function means we'll not risk missing the call to PopulateNil. WDYT?

Reply:
I don't think there is a good optimized way to do that since Reslice is called for both pointers and strings and checking for nil/empty will not be that straightforward with generics. Additionally, populate nil functionality doesn't fit properly in reslice IMO... it is too confusing with what it is supposed to do.

Reply:
An alternative way could be to have one method `ResliceAndPopulateNil` for pointers and another `Reslice` as a generic... that might turn out to be a bit better. I will do that in a bit today or tomorrow and see how it turns out.

Reply:
Oh, now I understand your point about pointers and strings, and how the generic types don't match. I'll approve this PR as is, and feel free to merge it if you can't figure out a better way.